examples/experimental/persistent-volume-provisioning/README.md

```diff
@@ -0,0 +1,141 @@
+--storage-class="foo/kubernetes.io/aws-ebs"
+```
+
+The CLI flag "--storage-class" can be used many times to add many provisioners. As the number and types of provisioners increase, admins
+can add many QoS classes with various backends to satisfy their needs.
+
+```
+--storage-class="foo/kubernetes.io/aws-ebs" 
+--storage-class="gold/kubernetes.io/aws-ebs-ssd"       # future implementation
+--storage-class="silver/kubernetes.io/aws-ebs-hd"      # future implementation
+--storage-class="bronze/kubernetes.io/aws-ebs-glacial" # future implementation
+```
+
+#### Provisioners
+
+Four Kubernetes persistent volume plugins have generic implementations for dynamic provisioning.
+
+The plugin names are:
+
+1. GCE PD  - "kubernetes.io/gce-pd"
+2. AWS EBS - "kubernetes.io/aws-ebs"
+3. OpenStack Cinder - "kubernetes.io/cinder"
+4. HostPath - "kubernetes.io/host-path"
+
+> Important! HostPath as a provisioner is meant for local development and testing only.  It _will not work_ in a multi-node cluster
+and is not supported in Kubernetes in any way outside local testing.
+
+Note that the first three are mutually independent and only 1 would be available at a time in a cloud environment.
+
+### User storage requests
+
+Users request dynamically provisioned storage by including a quality-of-service class in their `PersistentVolumeClaim`.
+An annotation is used to access this experimental feature.  In the future, the QoS class request will likely be a field on the claim itself.
+
+Assuming an admin configures a "foo" provisioner, a user would create a PVC with a special annotation requesting a quality-of-service class:
+
+```
+{
+  "kind": "PersistentVolumeClaim",
+  "apiVersion": "v1",
+  "metadata": {
+    "name": "claim1",
+    "annotations": {
+      "volume.experimental.kubernetes.io/quality-of-service": "foo"
+    }
+  },
+  "spec": {
+    "accessModes": [
+      "ReadWriteOnce"
+    ],
+    "resources": {
+      "requests": {
+        "storage": "3Gi"
+      }
+    }
+  }
+}
+```
+
+### Sample output
+
+This example uses HostPath but any provisioner would follow the same flow.
+
+First we note there are no Persistent Volumes in the cluster.  After creating a claim, we see a new PV is created
+and automatically bound to the claim requesting storage.
+
+
+``` 
+$ kubectl get pv
+
+$ k create -f examples/experimental/persistent-volume-provisioning/claim1.yaml 
+I1012 13:07:57.666759   22875 decoder.go:141] decoding stream as JSON
+persistentvolumeclaim "claim1" created
+
+$ kubectl get pv
+NAME                LABELS                                   CAPACITY   ACCESSMODES   STATUS    CLAIM            REASON    AGE
+pv-hostpath-r6z5o   createdby=hostpath-dynamic-provisioner   3Gi        RWO           Bound     default/claim1             2s
+
+$ kubectl get pvc
+NAME      LABELS    STATUS    VOLUME              CAPACITY   ACCESSMODES   AGE
+claim1    <none>    Bound     pv-hostpath-r6z5o   3Gi        RWO           7s
+
+```
+
+
```

Comment:
Should there be an example for `kubectl delete pvc` to explicitly note that the volumes are dynamically removed too?